src/twitchLanguageServer.ts

```diff
@@ -80,61 +80,56 @@ function onTtvChatMessage(channel: string, user: any, message: string) {
   parseMessage(userName, message);
 }
 
-let highlighterCommands = ['!line', '!highlight'];
-let highlightCommandUsed: string;
-
 function parseMessage(userName: string, message: string) {
-  message = message.toLocaleLowerCase();
-  // Note: as RamblingGeek suggested might want to look into
-  // using switch instead of if/else for better performance
-  if (!isHighlightCommand(message)) {
-    return;
-  }
-
-  const chatMessageRawAction = message
-    .slice(highlightCommandUsed.length)
-    .trim();
-
-  const messageParts = chatMessageRawAction.split(' ');
-  if (messageParts.length === 0) {
-    // Example: !<command>
-    return;
-  }
-
-  const notificationType = messageParts[0].startsWith('!')
-    ? 'unhighlight'
-    : 'highlight';
-  const lineNumber = messageParts[0].replace('!', '');
-  // Possible formats to support:
-  // !<command> <line number> <default to currently open file>
-  // !<command> <line number> <filename.ts>
-  // !<command> <line number> <filename.ts>
-  // !<command> <line number> <filename.ts>
-  // !<command> <line number> <filename.ts>
-  // !<command> !8 <filename.ts>
-  if (messageParts.length === 1) {
-    // Example: !<command> <line number>
-    connection.sendNotification(notificationType, {
-      line: +lineNumber,
-      twitchUser: userName
-    });
-  } else {
-    // Format Example: !<command> <line number> <filename.ts>
-    // Other Example: !<command> <line number> <filename.ts> <color>
-    connection.sendNotification(notificationType, {
-      line: +lineNumber,
-      filename: messageParts[1],
-      twitchUser: userName
-    });
-  }
-}
 
-function isHighlightCommand(message: string) {
-  return highlighterCommands.some(
-    (command: string): boolean => {
-      const comparison = message.startsWith(command.toLowerCase());
-      highlightCommandUsed = comparison ? command : '';
-      return comparison;
+  /**
+   * Regex pattern to verify the command is a highlight command
+   * groups the different sections of the command.
+   *
+   * See `https://regexr.com/48gf0` for my tests on the pattern.
+   *
+   * Matches:
+   *
+   * !line 5
+   * !line !5
+   * !line 5-10
+   * !line !5-15
+   * !line settings.json 5
```

Comment:
Haven't had a chance to run the code yet to test it but does this also support `!line 5 settings.json`? I only ask because it feels like that may be the natural/organic way to type/think about the usage. Do you agree? Might just be me who thinks that way though.

In general, how would we think/speak what we want to share? Something like "There's an issue on line 5 in your settings.json file"? Or "There's an issue in your settings.json file on line 5"? Should we support both or just one?

Reply:
Right now it would not support `!line 5 settings.json` but I can adjust the regex pattern to support that flow. See [https://regexr.com/48gf0](https://regexr.com/48gf0) to test the pattern.

Reply:
Just pushed a new commit which supports @clarkio's suggestion